test/demo/createToast.inttest.ts

```diff
@@ -0,0 +1,83 @@
+import 'jest';
+
+import {Application, Identity} from 'hadouken-js-adapter';
+
+import {NotificationOptions} from '../../src/client';
+
+import * as notifsRemote from './utils/notificationsRemoteExecution';
+import {isCenterShowing, getCardsByNotification, assertDOMMatches} from './utils/notificationCenterUtils';
+import {delay} from './utils/delay';
+import {getToastWindow, getToastCards} from './utils/toastUtils';
+import {createApp} from './utils/spawnRemote';
+import {assertNotificationStored} from './utils/storageRemote';
+
+const validOptions: NotificationOptions = {
+    body: 'Test Notification Body',
+    title: 'Test Notification Title'
+};
+
+const testManagerIdentity = {uuid: 'test-app', name: 'test-app'};
+
+describe('When calling createNotification', () => {
+    describe('With the notification center not showing', () => {
+        beforeAll(async () => {
+            // Hide the center to be sure we get toasts
+            if (await isCenterShowing()) {
+                await notifsRemote.toggleNotificationCenter(testManagerIdentity);
+            }
+        });
+
+        let testApp: Application;
+        let testWindowIdentity: Identity;
+        beforeEach(async () => {
+            testApp = await createApp(testManagerIdentity, {});
+            testWindowIdentity = await testApp.getWindow().then(w => w.identity);
+        });
+
+        afterEach(async () => {
+            await notifsRemote.clearAll(testWindowIdentity);
+            await testApp.quit();
+        });
```

Comment:
There should be some test that overwrites an existing toast with the same ID and check to see that the old toast has been removed and replaced.

Reply:
It would be a bit of a pain to properly test the removed/replaced for the toast window itself but could easily change the parameters and make sure that the new info shows up.